packages/tasit-action/src/TasitAction.js

```diff
@@ -180,6 +182,19 @@ class TransactionSubscription extends Subscription {
           return;
         }
 
+        setTimeout(() => {
```

Comment:
Hmm, but where's the part where we remove the earlier timeout that was set? Isn't that still missing? I think we might need to store a reference to each timeout that was set?

Reply:
Oh yes. You're right.
The code below makes sense?

```javascript
        if (this.#timer) clearTimeout(this.#timer);

        this.#timer = setTimeout(() => {
          const currentTime = Date.now();
          const timedOut =
            currentTime - this.#lastConfirmationTime >= this.getEventsTimeout();

          if (timedOut) {
            this._emitErrorEvent(
              new Error(`Event ${eventName} reached timeout.`),
              eventName
            );
          }
        }, this.getEventsTimeout());
```

Should timer be removed on `off()/usubscribe()` methods either?

Reply:
Yes, that code looks good. I think yes, off from confirmations for the transaction and unsubscribe should both clear it

Reply:
I've realized that the way I've written above the `#timer` field will belong to `TransactionSubscription` class and it isn't right because each timer must belong to their eventListener.
We could keep it that way because now we have only the "confirmation" event. 
But has another issue: Keeping that way it would be necessary to push down `unsubscribe()` and `off()` methods from `Subscription` superclass.
Because these two situations, I've decided to store the timer on eventListener Map and create these two new methods: `Subscription._setEventTimer(eventName, timer)` and `Subscription._clearEventTimerIfExists(eventName)` what do you think about decision @pcowgill ?

Note: I've just committed this change.

Reply:
> So if a user calls unsubscribe without clearing event timers manually, are the event timers still cleared?

Yes, that's the idea.

Reply:
Resolving this